File: qlty-cli/tests/lang/kotlin/basic.in/cyclomatic.kt
fun f1() {}

fun f2(foo: Boolean) {
    if (foo) {}
}

fun f3(foo: Boolean, bar: Boolean) {
    if (foo && bar) {}
}

fun f4() {
    while (true) {}
}

fun f5(x: Boolean) {
    val y: Int

    if (x) {
        y = 1
    } else {
        y = 2
    }
}

fun f6(x: Boolean, z: Boolean) {
    val y: Int

    if (x) {
        y = 1
    } else if (z) {
        y = 2
    }
}

fun f7(x: Boolean, z: Boolean) {
    val y: Int
    if (x) {
        y = 1
    } else if (z) {
        y = 2
    } else {
        y = 3
    }
}

fun f8() {
    val animals = arrayOf("dog", "cat", "bear")
    for (animal in animals) {
        println(animal)
    }
}

fun f9() {
    val animals = listOf("dog", "cat", "bear")
    animals.map(fun(animal: String): String {
        println(animal)
        return animal
    })
}

fun f10() {
    val animals = listOf("dog", "cat", "bear", "tiger")

    val filteredAnimals = animals.filter { it.length > 3 }
    for (animal in filteredAnimals) {
        println(animal)
    }

    if ("cat" in animals) {
        println("Found a cat!")
    }
}

File: qlty-cli/tests/lang/rust/basic.in/cyclomatic.rs
fn f1() {}

fn f2() {
    if foo {}
}

fn f3() {
    if foo && bar {}
}

fn f4() {
    while true {}
}

fn f5() {
    let y;

    if x {
        y = 1;
    } else {
        y = 2;
    }
}

fn f6() {
    let y;

    if x {
        y = 1;
    } else if z {
        y = 2;
    }
}

fn f7() {
    let y;
    if x {
        y = 1;
    } else if z {
        y = 2;
    } else {
        y = 3;
    }
}

fn f8() {
    for animal in &["dog", "cat", "bear"] {
        println!("{}", animal);
    }
}

fn f9() {
    let animals = vec!["dog", "cat", "bear"];
    let _ = animals
        .iter()
        .map(|animal| {
            println!("{}", animal);
            animal
        })
        .collect::<Vec<_>>();
}

fn f10() {
    let animals = vec!["dog", "cat", "bear", "tiger"];

    let filtered_animals: Vec<_> = animals.iter().filter(|&&animal| animal.len() > 3).collect();
    for animal in &filtered_animals {
        println!("{}", animal);
    }

    if animals.contains(&"cat") {
        println!("Found a cat!");
    }
}
